lib/omnibus/packagers/windows_base.rb

```diff
@@ -102,41 +107,41 @@ def timestamp_servers
       signing_identity[:timestamp_servers]
     end
 
+    def keypair_alias
+      signing_identity[:keypair_alias]
+    end
+
     def machine_store?
       signing_identity[:machine_store]
     end
 
-    #
-    # Iterates through available timestamp servers and tries to sign
-    # the file with with each server, stopping after the first to succeed.
-    # If none succeed, an exception is raised.
-    #
+    # signs the package with the given certificate
     def sign_package(package_file)
-      success = false
-      timestamp_servers.each do |ts|
-        success = try_sign(package_file, ts)
-        break if success
-      end
-      raise FailedToSignWindowsPackage.new unless success
+      raise FailedToSignWindowsPackage.new unless is_signed?(package_file)
     end
 
-    def try_sign(package_file, url)
+    def is_signed?(package_file)
       cmd = [].tap do |arr|
-        arr << "signtool.exe"
-        arr << "sign /v"
-        arr << "/t #{url}"
-        arr << "/fd #{algorithm}"
-        arr << "/sm" if machine_store?
-        arr << "/s #{cert_store_name}"
-        arr << "/sha1 #{thumbprint}"
-        arr << "/d #{project.package_name}"
-        arr << "\"#{package_file}\""
+        arr << "smctl.exe"
```

Comment:
I noticed that smctl supports different options, including a signing algorithm.  Is there a reason to remove this capability here?  Do we have any information on projects that might be using `algorithm` now that might be impacted?

Reply:
As far as I am aware, all windows msi signed packages were using this tool. If we want to use a signing algorithm, we would have to work that in as a new parameter. However, it looks like all previous msi packages that were signed, were using the same chef code sign ssl, and were signing the same way.